### PR TITLE
meson: use major/minor only in the lib soversion

### LINKED ISSUE #2290: Libs should have the right soversion
See https://github.com/radareorg/radare2/issues/18621 for more details.

### PR DESCRIPTION
**Pull request by ret2libc**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Before:
```
lrwxrwxrwx. 1 rschiron rschiron       23 May  9 17:09 /home/rschiron/.local/lib64/librz_type.so -> librz_type.so.0.4.0-git                                                                                        
-rwxr-xr-x. 1 rschiron rschiron  1269744 May  9 17:09 /home/rschiron/.local/lib64/librz_type.so.0.4.0-git
```
After:
```
lrwxrwxrwx. 1 root root   17 May 10 10:16 /lib/x86_64-linux-gnu/librz_type.so -> librz_type.so.0.4
lrwxrwxrwx. 1 root root   19 May 10 10:16 /lib/x86_64-linux-gnu/librz_type.so.0.4 -> librz_type.so.0.4.1
-rwxr-xr-x. 1 root root 2.0M May 10 10:16 /lib/x86_64-linux-gnu/librz_type.so.0.4.1
-rwxr-xr-x. 1 root root 2.0M May 10 10:12 /lib/x86_64-linux-gnu/librz_type.so.0.4.1-git
```

**Test plan**

1) install rizin from this PR
2) build a simple program against rz_util
3) run it, it should work
4) uninstall rizin; update the version field of the rizin directory; re-install it.
5) run again the test program and it should work, now calling the newer rz_util lib version

**Closing issues**

Closes #2290

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -1,5 +1,5 @@
 project('rizin', 'c',
-  version: 'v0.4.0-git',
+  version: 'v0.4.1',
```
Comment (XVilka) [inline meson.build]: Is this change necessary?

**Review by XVilka — APPROVED** [review]
LGTM apart from that version change

File: meson.build (hunk 1 above)
Comment (ret2libc) [reply meson.build]: no, not this one. `v0.4.0` is necessary though
